From a6d461bc7e8990c5b3571bba227b38c3deb1dd1e Mon Sep 17 00:00:00 2001
From: Finley Xiao <finley.xiao@rock-chips.com>
Date: Thu, 17 Aug 2017 09:53:18 +0800
Subject: [PATCH] arm64: dts: rockchip: rk3399: add gpu pvtm voltage table

stress test:
1. Antutu, use governor simpleondemand
2. Need for Speed, use governor simpleondemand
3. Glmark2, use userspace, scanning frequency

Change-Id: Ibe27380e582b193d900b0d55da3567ce553c32df
Signed-off-by: Finley Xiao <finley.xiao@rock-chips.com>
---
 arch/arm64/boot/dts/rockchip/rk3399-opp.dtsi | 36 ++++++++++++++++++++
 1 file changed, 36 insertions(+)

diff --git a/arch/arm64/boot/dts/rockchip/rk3399-opp.dtsi b/arch/arm64/boot/dts/rockchip/rk3399-opp.dtsi
index 0e58ae354664..e8955c13b4ac 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-opp.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3399-opp.dtsi
@@ -224,25 +224,61 @@
 	gpu_opp_table: opp-table2 {
 		compatible = "operating-points-v2";
 
+		rockchip,pvtm-voltage-sel = <
+			0        121000   0
+			121001   125500   1
+			125501   128500   2
+			128501   999999   3
+		>;
+		rockchip,pvtm-freq = <200000>;
+		rockchip,pvtm-volt = <900000>;
+		rockchip,pvtm-ch = <3 0>;
+		rockchip,pvtm-sample-time = <1000>;
+		rockchip,pvtm-number = <10>;
+		rockchip,pvtm-error = <1000>;
+		rockchip,pvtm-ref-temp = <41>;
+		rockchip,pvtm-temp-prop = <46 12>;
+		rockchip,pvtm-thermal-zone = "gpu-thermal";
+
 		opp-200000000 {
 			opp-hz = /bits/ 64 <200000000>;
 			opp-microvolt = <800000>;
+			opp-microvolt-L0 = <800000>;
+			opp-microvolt-L1 = <800000>;
+			opp-microvolt-L2 = <800000>;
+			opp-microvolt-L3 = <800000>;
 		};
 		opp-300000000 {
 			opp-hz = /bits/ 64 <300000000>;
 			opp-microvolt = <800000>;
+			opp-microvolt-L0 = <800000>;
+			opp-microvolt-L1 = <800000>;
+			opp-microvolt-L2 = <800000>;
+			opp-microvolt-L3 = <800000>;
 		};
 		opp-400000000 {
 			opp-hz = /bits/ 64 <400000000>;
 			opp-microvolt = <825000>;
+			opp-microvolt-L0 = <825000>;
+			opp-microvolt-L1 = <825000>;
+			opp-microvolt-L2 = <800000>;
+			opp-microvolt-L3 = <800000>;
 		};
 		opp-600000000 {
 			opp-hz = /bits/ 64 <600000000>;
 			opp-microvolt = <925000>;
+			opp-microvolt-L0 = <925000>;
+			opp-microvolt-L1 = <925000>;
+			opp-microvolt-L2 = <900000>;
+			opp-microvolt-L3 = <900000>;
 		};
 		opp-800000000 {
 			opp-hz = /bits/ 64 <800000000>;
 			opp-microvolt = <1100000>;
+			opp-microvolt-L0 = <1100000>;
+			opp-microvolt-L1 = <1075000>;
+			opp-microvolt-L2 = <1050000>;
+			opp-microvolt-L3 = <1025000>;
 		};
 	};
 
-- 
2.35.3

